### PR TITLE
feat: rename arguments of `$unnest()`

### LINKED ISSUE #1080: `<DataFrame>$unnest()`'s arguments are different from Python Polars

### PR DESCRIPTION
**Pull request by etiennebacher**
Close #1080

Also improve test coverage, and other docs changes due to roxygen2 update

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -6,12 +6,20 @@
...
+  and replaced by `...`. This doesn't change the previous behavior, e.g.
+  `df$unnest(names = c("a", "b"))` still works (#1170).
+
 ### Bug fixes
 
 - `$describe_plan()` and `$describe_optimized_plan()` are now consistent in their
   output. Previously, the former would return a Result-type output and the other
   would return nothing (as expected). They now both return nothing (#1175).
 
+
```
Comment (eitsupi) [inline NEWS.md:22]:
```suggestion
```

File: NEWS.md [hunk 2]
```diff
@@ -6,12 +6,20 @@
 
 - New method `$str$extract_many()` (#1163).
 
+
```
Comment (eitsupi) [inline NEWS.md:9]:
```suggestion
```

**Review by eitsupi — CHANGES_REQUESTED** [review]
I think this is a breaking change, so can you mark it as a breaking change?

**Comment by etiennebacher** [comment]
I don't think it is, the old behavior still works (and I added a test for this)

**Review by eitsupi — APPROVED** [review]
I see, thanks!

**Pull request merged**